### PR TITLE
fix ambigous std::abs overloads

### PR DESCRIPTION
**Pull request by ftalbrecht**

### CODE REVIEW

**Comment by ftalbrecht** [comment]
any objections against merging, @renefritze?

**Comment by renefritze** [comment]
Sorry, apparently I forgot to submit my review form a couple of days back.

File: dune/xt/common/fix-ambiguous-std-math-overloads.hh [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+#include <stdexcept>
+#include <cstdint>
+
+namespace Dune {
+
+
+// forwards, include are below
+template <int k>
+class bigunsignedint;
```
[the remark this answers is not in the document]
Comment (ftalbrecht) [reply dune/xt/common/fix-ambiguous-std-math-overloads.hh]: I was under the impression that in C++, something like line 55 would not compile if bigunsignedint was not declared before. But could be I am wrong...

File: dune/xt/common/fix-ambiguous-std-math-overloads.hh [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+Dune::bigunsignedint<k> abs(const Dune::bigunsignedint<k>& value)
+{
+  return value;
+}
+
+
+template <int k>
+inline Dune::bigunsignedint<k> pow(Dune::bigunsignedint<k> /*value*/, std::uintmax_t /*n*/)
+{
+  // do not pull in any DUNE header for DUNE_THROW
```
[the remark this answers is not in the document]
Comment (ftalbrecht) [reply dune/xt/common/fix-ambiguous-std-math-overloads.hh]: I thought that from the `config.h` we should not pull in too many headers. In particular: if we pull any DUNE header from within the `config.h`, any defines in the `config.h` from dependent dune modules (which will then be below the include) will not be visible to these headers. If we do not care about that, things would get easier here...

File: config.h.cmake [hunk 3]
```diff
@@ -211,6 +211,8 @@
   #endif // TBB_INTERFACE_VERSION < 4400
 #endif // HAVE_TBB
 
+// This is an unfortunate hack, see the header for an explanation.
+#include <dune/xt/common/fix-ambiguous-std-math-overloads.hh>
```
[the remark this answers is not in the document]
Comment (ftalbrecht) [reply config.h.cmake]: Yup. This pr arose from a compiler error I got in dune-gdt. This pr fixes that and in particular lets all xt modules as well as gdt compile...

File: dune/xt/common/fix-ambiguous-std-math-overloads.hh [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+inline bool isinf(Dune::bigunsignedint<k> /*value*/)
+{
+  return false;
+}
+
+
+} // namespace std
+
+
+#include <dune/common/bigunsignedint.hh>
```
[the remark this answers is not in the document]
Comment (ftalbrecht) [reply dune/xt/common/fix-ambiguous-std-math-overloads.hh]: So might as well do that above and have a proper exception, you are saying?

File: dune/xt/common/fix-ambiguous-std-math-overloads.hh [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+#include <cstdint>
+
+namespace Dune {
+
+
+// forwards, include are below
+template <int k>
+class bigunsignedint;
+
+class NotImplemented;
```
[the remark this answers is not in the document]
Comment (ftalbrecht) [reply dune/xt/common/fix-ambiguous-std-math-overloads.hh]: No, probably forgot that.

File: dune/xt/common/fix-ambiguous-std-math-overloads.hh (hunk 1 above)
[the remark this answers is not in the document]
Comment (renefritze) [reply dune/xt/common/fix-ambiguous-std-math-overloads.hh]: I meant why not move up the include instead.

File: dune/xt/common/fix-ambiguous-std-math-overloads.hh (hunk 2 above)
[the remark this answers is not in the document]
Comment (renefritze) [reply dune/xt/common/fix-ambiguous-std-math-overloads.hh]: Well, config.h pulls in lots of grid headers with the grid selector mechanism, right?
Wouldn't these kinds of defines be a design flaw anyhow? If a header in xt-common depends on defines from xt-grid I would consider that fix-worthy.

File: config.h.cmake (hunk 3 above)
[the remark this answers is not in the document]
Comment (renefritze) [reply config.h.cmake]: I didn't mean this particular problem, but solving another one by placing a header to be included in config.h.

File: dune/xt/common/fix-ambiguous-std-math-overloads.hh (hunk 4 above)
[the remark this answers is not in the document]
Comment (renefritze) [reply dune/xt/common/fix-ambiguous-std-math-overloads.hh]: Yes.

**Comment by ftalbrecht** [comment]
Applied the changes you proposed

**Pull request merged**